src/main/groovy/org/transmartproject/batch/patient/GatherCurrentPatientsReader.groovy

```diff
@@ -64,9 +69,17 @@ class GatherCurrentPatientsReader implements ItemStreamReader<Patient> {
     @SuppressWarnings('UnusedPrivateMethodParameter')
     private Patient mapRow(ResultSet rs, int rowNum) throws SQLException {
         String id = rs.getString(2)["$studyId:".length()..-1]
-        new Patient(
+        def patient = new Patient(
                 id: id,
                 code: rs.getLong(1))
+
+        patient.putDemographicValues([
+                (DemographicVariable.RACE): rs.getString('race_cd'),
+                (DemographicVariable.GENDER): rs.getString('sex_cd'),
+                (DemographicVariable.AGE): rs.getObject('age_in_years_num'),
+        ])
```

Comment:
I don't think this is correct. If someone deletes the race of a patient, it won't be deleted from the database. Better even not to fetch this data.